spring-boot-starter-rocketmq/src/main/java/org/apache/rocketmq/spring/starter/core/DefaultRocketMQListenerContainer.java

```diff
@@ -0,0 +1,293 @@
+@Slf4j
+public class DefaultRocketMQListenerContainer implements InitializingBean, RocketMQListenerContainer {
+
+    @Setter
+    @Getter
+    private long suspendCurrentQueueTimeMillis = 1000;
+
+    /**
+     * Message consume retry strategy<br> -1,no retry,put into DLQ directly<br> 0,broker control retry frequency<br>
+     * >0,client control retry frequency
+     */
+    @Setter
+    @Getter
+    private int delayLevelWhenNextConsume = 0;
+
+    @Setter
+    @Getter
+    private String consumerGroup;
+
+    @Setter
+    @Getter
+    private String nameServer;
+
+    @Setter
+    @Getter
+    private String topic;
+
+    @Setter
+    @Getter
+    private ConsumeMode consumeMode = ConsumeMode.CONCURRENTLY;
+
+    @Setter
+    @Getter
+    private SelectorType selectorType = SelectorType.TAG;
+
+    @Setter
+    @Getter
+    private String selectorExpress = "*";
+
+    @Setter
+    @Getter
+    private MessageModel messageModel = MessageModel.CLUSTERING;
+
+    @Setter
+    @Getter
+    private int consumeThreadMax = 64;
+
+    @Getter
+    @Setter
+    private String charset = "UTF-8";
+
+    @Setter
+    @Getter
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Setter
+    @Getter
+    private boolean started;
+
+    @Setter
+    private RocketMQListener rocketMQListener;
+
+    private DefaultMQPushConsumer consumer;
+
+    private Class messageType;
+
+    public void setupMessageListener(RocketMQListener rocketMQListener) {
+        this.rocketMQListener = rocketMQListener;
+    }
+
+    @Override
+    public void destroy() throws Exception {
+        this.setStarted(false);
+        if (Objects.nonNull(consumer)) {
+            consumer.shutdown();
+        }
+        log.info("container destroyed, {}", this.toString());
+    }
+
+    public synchronized void start() throws MQClientException {
+
+        if (this.isStarted()) {
+            throw new IllegalStateException("container already started. " + this.toString());
+        }
+
+        initRocketMQPushConsumer();
+
+        // parse message type
+        this.messageType = getMessageType();
+        log.debug("msgType: {}", messageType.getName());
+
+        consumer.start();
+        this.setStarted(true);
+
+        log.info("started container: {}", this.toString());
+    }
+
+    public class DefaultMessageListenerConcurrently implements MessageListenerConcurrently {
+
+        @SuppressWarnings("unchecked")
+        public ConsumeConcurrentlyStatus consumeMessage(List<MessageExt> msgs, ConsumeConcurrentlyContext context) {
+            for (MessageExt messageExt : msgs) {
+                log.debug("received msg: {}", messageExt);
+                try {
+                    long now = System.currentTimeMillis();
+                    rocketMQListener.onMessage(doConvertMessage(messageExt));
+                    long costTime = System.currentTimeMillis() - now;
+                    log.info("consume {} cost: {} ms", messageExt.getMsgId(), costTime);
```

Comment:
Hi, it's not a good idea to log trace info here, it may bring performance loss, let users do this work in their own `onMessage` callback.

Reply:
thanks for your suggestion, I changed log level to `debug`. In many cases, developers not do this in their own implementation, but expect see cost time.